proseco/gui_char.py

```diff
@@ -0,0 +1,1629 @@
+                [-2.7052897180601e-09,-5.3570209777220199e-09],
+                [9.7557263803716508e-10,1.91073314224894e-08],
+                [-2.9486515531677799e-08,-4.8576658185286602e-09],
+                [8.3119801831277497e-13,2.0109207042849501e-10],
+                [-1.96043819238097e-10,5.0972154587641396e-16],
+                [5.1413424477146302e-13,1.99339355492595e-10],
+                [-1.97282476269237e-10,2.5273983431918399e-14]
+                ],
+            "Pix2Arc": 4.9628899999999998,
+            "CatalogRadius": 1.1000000000000001,
+            "CatalogMinMag": 14.5,
+            "ManvrErrorSigmaInit": [4.8481368110953597e-05,4.8481368110953605e-07,4.8481368110953605e-07],
+            "ManvrErrorSearchBoxMargin": 9.6962736221907193e-05,
+            "BadStarList": [36178592,39980640,185871616,188751528,190977856,260968880,260972216,261621080,296753512,300948368,301078152,301080376,301465776,335025128,335028096,414324824,444743456,465456712,490220520,502793400,509225640,570033768,614606480,637144600,647632648,650249416,656409216,690625776,692724384,788418168,849226688,956175008,989598624,1004817824,1016736608,1044122248,1117787424,1130635848,1130649544,1161827976,1196953168,1197635184],
+            "ACAFaintCommandLimit": 13.98,
+            "minCmdMagErrorBar": 1.5,
+            "StarAcqTime": 270,
+            "ERImageSize": 2,
+            "Dark_Map": {
+                "Scale_4c": 1.5900000000000001
+                }
+            },
+        "Guide": [
+            {
+                "SearchSettings": {
+                    "DoColumnRegisterCheck": 1,
+                    "DoBminusVcheck": 1
+                    },
+                "Inertial": {
+                    "MagLimit": [5.7999999999999998,10.199999999999999],
+                    "MagErrorTol": 100,
+                    "PosErrorTol": 3000,
+                    "ASPQ1Lim": [0,0],
+                    "ASPQ2Lim": [0,0],
+                    "ASPQ3Lim": [0,999],
+                    "VARIABLELim": -9999,
+                    "MagErrRand": 0.26000000000000001,
+                    "MagErrSyst": 0,
+                    "MaxMagError": 1.5
+                    },
+                "Body": {
+                    "Column": {
+                        "MagDiff": 5,
+                        "Separation": 4
+                        },
+                    "Register": {
+                        "MagDiff": 5,
+                        "Separation": 4,
+                        "Width": 2
+                        },
+                    "Pixels": {
+                        "ZPixLim": [-512.5,511.5],
+                        "YPixLim": [-512.5,511.5],
+                        "Center": [-0.5,-0.5],
+                        "EdgeBuffer": 5,
+                        "BadPixels": [
+                            [-245,0,454,454],
+                            [-511,-511,9,9],
+                            [-510,-510,-289,-289],

```

Comment:
Should chop this down to actual bad pixels (i.e. no hot pixels in this list), and dynamically generate hot pixel list from 1 or more dark maps.  This would be from dark maps taken from before the obs date.  The "or more" option refers to dynamically replicating the current hot pixel algorithm using the 3 readouts before the date.

Eventually the whole hot pixel list will get scrapped, but in the meantime a dynamic hot pixel list will provide continuity with the existing algorithm.

Reply:
Thought not sure on timetable for eventually coming back to eliminate the stage-based star selection to know how much time/effort to allocate to moderate improvements in this code.

Reply:
I don't think you need to worry about matching SAUSAGE.  My understanding was that you had done reasonable matching validation for ACA LTS.  But more to the point, Sabina already agreed that we should just push forward with any obvious improvements now given that there will be significant testing of the output catalogs in conjunction with acq star selection.

Reply:
I'm not sure about timetable either, except that it will be Pi times longer than any reasonable expectation.  I have a feeling that if you can make some low-hanging fruit improvements then this may be good enough for a while and allow me to focus (with @malgosias) on patch development.

Reply:
I would say to start with the baby step because it is very easy and requires only a small new function to dynamically add to the bad pixels list.

The imposter-like approach is a fundamental change to the algorithm.  Remember also that currently there may be as much a problem from warm/hot pixels in the 8 background pixels as from centroid disturbances, and those issues cannot be easily detected without doing annie-like tracking simulation.

Reply:
Right.  I will definitely do the baby step.  If we have time/inclination for more we can look at more steps.

I think that allowing us to select 6.0 mag stars that are presently being excluding due to the presence of 2000 e-/sec hot pixels would be a step in the right direction even if the selection would not be robust to all problems (such as background issues).  The current algorithm is also not robust to all problems.

Reply:
Regarding "I don't think you need to worry about matching SAUSAGE."

Agreed.  I just thought for my process that it made a lot of sense to baseline the "as close to SAUSAGE as we're going to get" that is what the code is basically already doing, and then have discrete improvements from that baseline in new PRs against this branch.

Reply:
But getting close to SAUSAGE at any point in this code is irrelevant since final testing will be completely independent of the original SAUSAGE selections (except to demonstrate that the new catalogs are better by some metric).  So you will spend time testing now (validating that it gets close to SAUSAGE), for no impact later.

Reply:
And yes, the dynamic hot pixel list should be mag dependent.